### PR TITLE
User can stream

### PR DESCRIPTION
**Pull request by humbertovnavarro**
Screenshot from 2021-09-09 11-44-02
Screenshot from 2021-09-09 11-44-25
Screenshot from 2021-09-09 11-51-25
Screenshot from 2021-09-09 11-51-02

### CODE REVIEW

**Comment by humbertovnavarro** [comment]
https://user-images.githubusercontent.com/16251746/132756913-d0dbaccd-45cd-4fae-999e-352d3d34dc14.mp4

File: client/app.jsx [hunk 1]
```diff
@@ -1,8 +1,48 @@
...
+    };
+  }
+
+  componentDidMount() {
+    window.addEventListener('hashchange', () => {
+      const route = parseRoute(window.location.hash);
+      this.setState({
+        route: route,
+        path: route.path,
+        params: route.params
```
Comment (thebearingedge) [inline client/app.jsx:20]: you can leave these in `route`, no?

File: client/app.jsx [hunk 2]
```diff
@@ -1,8 +1,48 @@
...
+        return <Channel />;
+    }
+  }
+
   render() {
-    return <Home />;
+    const contextValue = {
+      route: this.state.route,
+      path: this.state.route.path,
+      params: this.state.route.params
```
Comment (thebearingedge) [inline client/app.jsx:39]: you can keep your code cleaner by just passing the `route` and not cluttering the context namespace

File: server/routes.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    }
+    if(fs.existsSync(path.join(__dirname,`/public/live/${id}.mpd`))) {
+      res.json({
+        isLive: true
+      });
+    } else {
+      res.json({
+        isLive: false
+      });
+    }
```
Comment (thebearingedge) [inline server/routes.js:34]: We don't use synchronous i/o in web servers.

**Review by thebearingedge — CHANGES_REQUESTED** [review]

**Review by thebearingedge — APPROVED** [review]

**Pull request merged**